core/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java

```diff
@@ -193,14 +320,200 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (nestedFilter != null) {
             builder.field("nested_filter", nestedFilter, params);
         }
-        if (coerce != null) {
-            builder.field("coerce", coerce);
-        }
-        if (ignoreMalformed != null) {
-            builder.field("ignore_malformed", ignoreMalformed);
-        }
+        builder.field("coerce", coerce);
+        builder.field("ignore_malformed", ignoreMalformed);
 
         builder.endObject();
         return builder;
     }
+
+    @Override
+    public String getWriteableName() {
+        return NAME;
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (super.equals(object)) {
+            return true;
```

Comment:
But then, if any of the superclasses did implement their own equals - wouldn't I actually want to check their equals implementation?

Can add the check for reference equality explicitly of course.

Somehow this discussion reminds me of https://en.wikipedia.org/wiki/Liskov_substitution_principle